pandas/core/reshape/merge.py

```diff
@@ -787,33 +787,49 @@ def _maybe_add_join_keys(self, result, left_indexer, right_indexer):
+                right_in, left_in = False, False
+                if not array_like:
+                    right_in = (
+                        name in self.right_on
+                        or self.right_index is True
+                        and name in self.right.index.names
+                    )
+                    left_in = (
+                        name in self.left_on
+                        or self.left_index is True
+                        and name in self.left.index.names
+                    )
+                if right_in and left_in or array_like or self.how == "asof":
 
-                if left_indexer is not None and right_indexer is not None:
-                    if name in self.left:
+                    if left_indexer is not None and right_indexer is not None:
+                        if name in self.left:
 
-                        if left_has_missing is None:
-                            left_has_missing = (left_indexer == -1).any()
+                            if left_has_missing is None:
+                                left_has_missing = (left_indexer == -1).any()
 
-                        if left_has_missing:
-                            take_right = self.right_join_keys[i]
+                            if left_has_missing:
+                                take_right = self.right_join_keys[i]
 
-                            if not is_dtype_equal(
-                                result[name].dtype, self.left[name].dtype
-                            ):
-                                take_left = self.left[name]._values
+                                if not is_dtype_equal(
+                                    result[name].dtype, self.left[name].dtype
+                                ):
+                                    take_left = self.left[name]._values
 
-                    elif name in self.right:
+                        elif name in self.right:
 
-                        if right_has_missing is None:
-                            right_has_missing = (right_indexer == -1).any()
+                            if right_has_missing is None:
+                                right_has_missing = (right_indexer == -1).any()
 
-                        if right_has_missing:
-                            take_left = self.left_join_keys[i]
+                            if right_has_missing:
+                                take_left = self.left_join_keys[i]
 
-                            if not is_dtype_equal(
-                                result[name].dtype, self.right[name].dtype
-                            ):
-                                take_right = self.right[name]._values
+                                if not is_dtype_equal(
+                                    result[name].dtype, self.right[name].dtype
+                                ):
+                                    take_right = self.right[name]._values
+                else:
+                    continue
```

Comment:
Is it possible to write this as
```python
if <condition>:
    continue
<keep existing code here as it is without having to indent it futher>
```
?

Reply:
@MarcoGorelli Yeah, changed it aroung. I think this is a bit more elegant too. I did it the other way round because imo the if condition is a bit more readable, but was split.. Maybe you could have a look again with the changed condition?

thx :)